lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -140,81 +175,68 @@ def test__has_access_descriptor(self):
         with self.assertRaises(ValueError):
             access._has_access_descriptor(user, 'not_load_or_staff', descriptor)
 
+    @ddt.data(
+        (True, None, access_response.VisibilityError),
+        (False, None),
+        (True, YESTERDAY, access_response.VisibilityError),
+        (False, YESTERDAY),
+        (True, TOMORROW, access_response.VisibilityError),
+        (False, TOMORROW, access_response.StartDateError)
```

Comment:
To user `itertools`, you may need to explicitly pass in `None` for the expected error.